src/HtmlProcessor/CssToAttributeConverter.php

```diff
@@ -20,7 +20,7 @@ class CssToAttributeConverter extends AbstractHtmlProcessor
      * only for certain values, the mapping is an object with a whitelist
      * of nodes and values.
      *
-     * @var mixed[][]
+     * @var array<string, array{attribute: string, nodes?: array<int, string>, values?: array<int, string>}>
```

Comment:
Oh, so the `?` goes *here* for an array element which *must either exist with the given type or not exist at all* (i.e. cannot be a `null` entry)?

Where do you get this information from?  I only found out about the `array{key: type}` syntax by seeing it used elsewhere and/or in the Psalm error report.  I've said it before and I'll say it again: "Documentation!" 😉

Reply:
I've found the contents of the Psalm documentation to be quite good, and its navigation to be an utter mess. I usually google what I need to find.

In this case, googling for "psalm arrays" lead me to this page explaining the syntax:
https://psalm.dev/docs/annotating_code/type_syntax/array_types/

Reply:
Actually the documenation is quite good.  It's just hard to navigate and search.  But so is the internet as a whole.  Often when I put "Psalm" in the search box, a lot of the results are about religion, even if I include "PHP".  But I think including `site:psalm.dev` in the search box is the way forward.  Thanks for the pointer :)

Reply:
Sorry, didn't properly read the first paragraph of your comment before replying.  But it seems we agree on both aspects :)

I prefer to use DuckDuckGo and fall back to Google if I can't find what I'm looking for with the former.  With DDG (and presumably also Bing, etc.), the top result for "psalm arrays" is "Topical Bible: Arrays - Arrays (1 Occurrence) Esther 6:8 let the royal apparel be brought with which the king arrayshimself, and the horse that the king rides upon, and on the head of which the royal crown is set;"  I won't post the link, but even by posting the snippet I am probably further confusing the AIs.